### ISSUE TITLE
[NFR] Update \Phalcon\Text::uncamelize() with optional delimiter to change default "_"

### ISSUE THREAD

**Issue by virgofx** [phalcon/cphalcon#10396]
The current delimiter in `string.h` uses an underscore:

`smart_str_appendc(&uncamelize_str, '_');`

However, it's very common to convert using hyphens (depending on the situation). Either way, just set an optional parameter in the Phalcon\Text class with the default to the current underscore (this way to prevent breaking any compatibility).

Here is what the updated relevant document change would look like. Just need to update zephir's `string.h` to handle the passed in parameter.

```php
    /**
     * Uncamelize strings which are camelized.
     *
     * <code>
     *   echo Phalcon\Text::uncamelize('CocoBongo'); //coco_bongo
     *   echo Phalcon\Text::uncamelize('CocoBongo', '-'); //coco-bongo
     * </code>
     *
     * @param string $str    The string to uncamelize
     * @param string $delimiter   Optional delimiter to use. By default this is "_"
     * @return string 
     */
	public static function uncamelize($str, $delimiter = '_') {}
```

**Comment by sergeyklay** [comment]
:+1:

**Comment by dreamsxin** [comment]
+1

**Comment by sergeyklay** [comment]
Implemented in the 2.1.x branch

**Comment by virgofx** [comment]
@sergeyklay Thank you :+1:
